src/app/shared/model/creative.model.ts

```diff
@@ -0,0 +1,77 @@
+import { BaseModel, HalDoc } from 'ngx-prx-styleguide';
+import { Observable } from 'rxjs/Observable';
+
+export class CreativeModel extends BaseModel {
+  public id: number;
+  public status: string;
+  public filename: string;
+  public zone: string;
+  public size: number;
+  public contentType: string;
+  public label: string;
+  public length: number;
+  public layer: number;
+  public bitRate: number;
+  public channelMode: string;
+  public uploadPath: string;
+  public format: string;
+
+  SETABLE = []; // TODO
+
+  VALIDATORS = {}; // TODO
+
+  constructor(campaign: HalDoc, creative: HalDoc, loadRelated = true) {
+    super();
+    this.init(campaign, creative, loadRelated);
+  }
+
+  key(): string {
+    if (this.doc) {
+      return `prx.creative.${this.doc.id}`;
```

Comment:
seems like you might want a ".new" key as well if you're doing local storage for new records before save, but I can also see where you're not quite at the point of creating these records in app yet

Reply:
good call!